cmd/registry/cmd/check/rules/rule0001/mime_type_contents.go

```diff
@@ -0,0 +1,74 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package rule0001
+
+import (
+	"fmt"
+	"mime"
+	"net/http"
+	"strings"
+
+	"github.com/apigee/registry/cmd/registry/cmd/check/lint"
+	"github.com/apigee/registry/rpc"
+)
+
+var mimeTypeContents = &lint.FieldRule{
+	Name: lint.NewRuleName(1, "mime-type-contents"),
+	OnlyIf: func(resource lint.Resource, field string) bool {
+		return field == "MimeType"
+	},
+	ApplyToField: func(resource lint.Resource, field string, value interface{}) []lint.Problem {
+		var declared string
+		var contents []byte
+		switch t := resource.(type) {
+		case *rpc.ApiSpec:
+			declared = t.GetMimeType()
+			contents = t.GetContents()
+		case *rpc.Artifact:
+			declared = t.GetMimeType()
+			contents = t.GetContents()
+		}
+		if len(contents) == 0 {
+			return nil
+		}
+		detected := http.DetectContentType(contents)
```

Comment:
I don't think this will work for us since we use some unregistered (`x.`) content types. Interesting idea, though.

Reply:
We could extend it to ignore such types.

Reply:
Also, note that we'll handle specific Hub attribute types in more specific rules.

Reply:
> Great start - I like the structure and derivations from API linter. I didn't try to diagnose the data race that's breaking the tests but would prefer that we wait to merge after that's fixed. Otherwise LGTM.

Fixed